src/OpenLoco/src/Input/Shortcuts.cpp

```diff
@@ -397,7 +409,39 @@ namespace OpenLoco::Input::Shortcuts
     // 0x004BF3DC
     static void sendMessage()
     {
-        call(0x004BF3DC);
+        if (isEditorMode())
+            return;
+
+        if (!isNetworked())
+            return;
+
+        if (isTitleMode())
+        {
+            auto* caller = WindowManager::find(WindowType::titleMenu);
+            if (caller == nullptr)
+                return;
+
+            WindowManager::close(WindowType::multiplayer);
+
+            FormatArguments args{};
+            args.push(StringIds::the_other_player);
+
+            const int callingWidget = 4;
+            Ui::Windows::TextInput::openTextInput(caller, StringIds::chat_title, StringIds::chat_instructions, StringIds::empty, callingWidget, const_cast<void*>(&args));
+        }
+        else
+        {
+            auto* caller = WindowManager::find(WindowType::timeToolbar);
+            if (caller == nullptr)
+                return;
+
+            const auto* opponent = CompanyManager::getOpponent();
+            FormatArguments args{};
+            args.push(opponent->name);
+
+            const int callingWidget = 2;
+            Ui::Windows::TextInput::openTextInput(caller, StringIds::chat_title, StringIds::chat_instructions, StringIds::empty, callingWidget, const_cast<void*>(&args));
```

Comment:
This should probably just call `TimePanel::beginSendChatMessage()`. Though this implementation seems more accurate; the existing implementation skips the opponent name.